src/modules/comments/dtos/comment.dtos.ts

```diff
@@ -8,7 +8,7 @@ export class CommentDto {
   readonly comment: string;
 
   @IsNotEmpty()
-  readonly user: MongooseSchema.Types.ObjectId;
+  readonly postId: MongooseSchema.Types.ObjectId;
```

Comment:
Con el postId tenemos como asignarle el comentario a dicho post, eso esta bien ¿pero como vamos a saber que usuario lo hizo?

Reply:
Eso ya esta resuleto, recuerda que tenemos en las request el usuario que nos regresa la validacion del jwt, y entre sus propiedades esta el id que retorna nuestra funcion de validate en el stategy del jwt

Reply:
` @Post()
  @HttpCode(HttpStatus.CREATED)
  @UseGuards(AuthGuard("jwt"))
  async createComment( @Body() comment: CommentDto, @Auth() {id}:User): Promise<IResponseJson<Comment>> {
    const commentNew = await this.commentService.create(comment, id);
    return {
      data: commentNew,
      message: "Comment created"
    };
  }`

Justo en esa parte con el custom decorator @Auth(), estamos recibiendo desde las request el usuario y ya lo pasamos al servicio, dado que de esa manera nos aseguramos de una manera mas fiable que el id siempre sea del autor que lo crea